terraform-aws-github-runner/modules/runners-instances/templates/user-data.sh

```diff
@@ -104,6 +104,63 @@ ${install_config_runner}
 retry sudo dnf groupinstall -y 'Development Tools'
 retry sudo dnf install -y "kernel-devel-uname-r == $(uname -r)" || true
 
+%{ if wiz_secrets_arn != null ~}
+# Install Wiz Sensor - a runtime security agent
+echo "Fetching Wiz secrets from AWS Secrets Manager"
+
+(
+    # Allow the script to continue even if the function fails, so that we gracefully handle
+    # wiz installation failures
+    set +e
+
+    # Function to get region from ARN
+    get_region_from_arn() {
+        local arn=$1
+        # Extract region from ARN (format: arn:aws:service:region:account:resource)
+        local region=$(echo "$arn" | cut -d':' -f4)
+        if [ -n "$region" ]; then
+            echo "$region"
+        else
+            echo ""
+        fi
+    }
+
+    SECRET_REGION=$(get_region_from_arn "${wiz_secrets_arn}")
+    if [ -z "$SECRET_REGION" ]; then
+        echo "Warning: Region is required in the Secrets Manager ARN. Skipping Wiz installation."
+        metric_report "linux_userdata.wiz_failure_arn_invalid" 1
+    else
+        WIZ_SECRET_RAW=$(retry aws secretsmanager get-secret-value --secret-id "${wiz_secrets_arn}" --region "$SECRET_REGION" --query 'SecretString' --output text)
+        if [ $? -eq 0 ] && [ ! -z "$WIZ_SECRET_RAW" ]; then
+          echo "Successfully retrieved Wiz secrets"
+          echo "Extracting Wiz runtime sensor credentials"
+          WIZ_SECRET_JSON=$(echo "$WIZ_SECRET_RAW" | tr -d '\n\r') # Remove newlines to fix malformed JSON (it's how it's stored in AWS Secrets Manager)
+          WIZ_API_CLIENT_ID=$(echo "$WIZ_SECRET_JSON" | jq -r '.WIZ_RUNTIME_SENSOR_CLIENT_ID // empty')
+          WIZ_API_CLIENT_SECRET=$(echo "$WIZ_SECRET_JSON" | jq -r '.WIZ_RUNTIME_SENSOR_CLIENT_SECRET // empty')
+          if [ ! -z "$WIZ_API_CLIENT_ID" ] && [ ! -z "$WIZ_API_CLIENT_SECRET" ]; then
+            echo "Installing Wiz runtime sensor"
+            if ! WIZ_API_CLIENT_ID="$WIZ_API_CLIENT_ID" WIZ_API_CLIENT_SECRET="$WIZ_API_CLIENT_SECRET" \
+            sudo -E bash -c "$(curl -L https://downloads.wiz.io/sensor/sensor_install.sh)"; then
```

Comment:
I wonder if it's better to host this shell script on our end instead of downloading it from `https://downloads.wiz.io/sensor/sensor_install.sh` all the times

* We can version track it
* Avoid flaky network issue
* Avoid someone tampering with it without us knowing

It's similar to how we host NVIDIA installation script on S3

Reply:
That's done in this PR: https://github.com/pytorch/test-infra/pull/6739